liblepton/scheme/conf/upgrade.scm

```diff
@@ -0,0 +1,233 @@
+;; Lepton EDA
+;; liblepton - Lepton's library - Scheme API
+;; Copyright (C) 2019 dmn <graahnul.grom@gmail.com>
+;; License: GPLv2+. See the COPYING file
+;;
+
+( define-module ( conf upgrade )
+
+  #:use-module  ( ice-9 format )
+  #:use-module  ( geda  config )
+  #:use-module  ( conf  keys   )
+
+  #:export      ( convert )
+)
+
+
+
+; public:
+;
+; convert(): upgrade gEDA configuration
+;
+; Read legacy gEDA configuration from geda.conf (in current
+; directory), geda-user.conf or geda-system.conf file,
+; convert it (using new names) and produce the corresponding
+; lepton*.conf configuration file.
+; We get conversion information (list of keys, old and new names)
+; from the [keys-list] defined in the (conf keys) module.
+;
+; [what]:                'local, 'user or 'system - config to convert
+; [report-absent-keys]:  print messages about missing keys in old cfg file
+; [overwrite]:           if #t, overwrite existing new cfg file
+; {thr}:                 'ctx 'infile 'load 'outfile 'save
+; {ret}:                 new cfg file full name or #f on failure
+;
+; This function might throw the following exceptions:
+; - 'ctx:      cannot get configuration context
+; - 'infile:   legacy config file does not exist
+; - 'load:     cannot read legacy config file
+; - 'outfile:  output config file already exists
+; - 'save:     cannot write output config file
+;
+( define* ( convert what #:key (report-absent-keys #f) (overwrite #f) )
+( let
+  (
+  ( cfg    #f )
+  ( fname  #f )
+  ( keys  '() )
+  ( data  '() )
+  )
+
+  ;
+  ; read legacy config file (geda*.conf):
+  ;
+
+  ( config-set-legacy-mode! #t ) ; use geda*.conf files
+
+  ( set! cfg ( get-cfg-ctx what ) )
```

Comment:
Defining local variables and re-setting them makes the code less readable in my view. Wouldn't it be better just to use `(let* ...) here?

Reply:
This function is called in two places to get different contexts. Similar assignments maintains the code symmetry, so to say, and makes it easier to understand, IMHO. :)